drjt/Animal/lib/Animal.pm

```diff
@@ -38,6 +38,77 @@ if you don't export anything, such as for a purely object-oriented module.
 
 =head1 SUBROUTINES/METHODS
 
+=head2 new
+
+Constructor, takes a name as a parameter
+
+=cut
+
+sub new {
+  my $class = shift;
+  my $name = shift;
+  my $self = { name => $name, colour => $class->default_colour() };
+  bless $self, $class;
+  return $self;
```

Comment:
You don't strictly need to have the object hash defined separately, unless you want to conduct additional processing after blessing it. You could simplify this to:
```perl
return bless { name => $name, colour => $class->default_colour() }, $class;
```